libpod/container_internal.go

```diff
@@ -2674,7 +2674,12 @@ func (c *Container) update(resources *spec.LinuxResources, restartPolicy *string
 		return fmt.Errorf("must provide restart policy if updating restart retries: %w", define.ErrInvalidArg)
 	}
 
-	oldResources := c.config.Spec.Linux.Resources
+	var oldResources *spec.LinuxResources
+	if c.config.Spec.Linux.Resources != nil {
+		if err := JSONDeepCopy(oldResources, c.config.Spec.Linux.Resources); err != nil {
+			return err
+		}
+	}
```

Comment:
I don't get that, you are copying a nil pointer on that which mean this makes no sense and nothing is copied AFIACT.
If think you want to copy the current settings to oldResources in that case you must change the JSONDeepCopy argument order unless I am missing something.

Reply:
Thanks, I missed it.